vscode-dotnet-runtime-library/src/Acquisition/DebianDistroSDKProvider.ts

```diff
@@ -0,0 +1,28 @@
+/* --------------------------------------------------------------------------------------------
+ *  Licensed to the .NET Foundation under one or more agreements.
+*  The .NET Foundation licenses this file to you under the MIT license.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import { ICommandExecutor } from '../Utils/ICommandExecutor';
+import { IUtilityContext } from '../Utils/IUtilityContext';
+import { GenericDistroSDKProvider } from './GenericDistroSDKProvider';
+import { IAcquisitionWorkerContext } from './IAcquisitionWorkerContext';
+import { DistroVersionPair } from './LinuxVersionResolver';
+
+export class DebianDistroSDKProvider extends GenericDistroSDKProvider
+{
+    constructor(distroVersion : DistroVersionPair, context : IAcquisitionWorkerContext, utilContext : IUtilityContext, executor : ICommandExecutor | null = null)
+    {
+        super(distroVersion, context, utilContext, executor);
+    }
+
+    protected myVersionDetails() : any
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        const distroVersions = this.distroJson[this.distroVersion.distro][this.distroVersionsKey];
+        const targetVersion = Math.floor(parseFloat(this.distroVersion.version[0])).toFixed(1);
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        const versionData = distroVersions.filter((x: { [x: string]: string; }) => x[this.versionKey] === String(targetVersion));
+        return versionData;
+    }
```

Comment:
Something I noticed, is that if we try to install 8 and then 9 on debian without restarting and very soon after each other, the 2nd attempt may fail because the dpkg lock is held. I think this is a broader issue for the entire codebase so it's not a blocker for your PR, but it may be worth looking into. One of those apt-get running processes doesn't seem to be dying: `dpkg: error: dpkg frontend lock was locked by another process with pid blah`

You can check the hanging process which is in the output with something like `ps -P pid`, which might be valuable information to get.

Reply:
This may actually have been triggered by the package check failure above.